core/src/main/java/org/testcontainers/containers/RemoteDockerImage.java

```diff
@@ -0,0 +1,92 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.model.Image;
+import com.github.dockerjava.core.command.PullImageResultCallback;
+import com.google.common.util.concurrent.Futures;
+import lombok.NonNull;
+import lombok.experimental.Delegate;
+import org.slf4j.LoggerFactory;
+import org.slf4j.profiler.Profiler;
+import org.testcontainers.DockerClientFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+public class RemoteDockerImage implements Future<String> {
+
+    private static final Set<String> AVAILABLE_IMAGE_NAME_CACHE = new HashSet<>();
+
+    @Delegate
+    private final Future<String> delegate;
+
+    public RemoteDockerImage(@NonNull String repository, String tag) {
```

Comment:
We decided to use two arguments instead of one to make it more explicit that it must contain both repository and tag.